samza-api/src/main/java/org/apache/samza/runtime/ApplicationRunner.java

```diff
@@ -79,13 +80,28 @@ public ApplicationRunner(Config config) {
   }
 
   /**
-   * Method to be invoked to deploy and run the actual Samza jobs to execute {@link StreamApplication}
+   * Deploy and run the Samza jobs to execute {@link StreamApplication}
    *
    * @param streamApp  the user-defined {@link StreamApplication} object
    */
   public abstract void run(StreamApplication streamApp);
 
   /**
+   * Kill the Samza jobs represented by {@link StreamApplication}
+   *
+   * @param streamApp  the user-defined {@link StreamApplication} object
+   */
+  public abstract void kill(StreamApplication streamApp);
+
+  /**
+   * Get the collective status of the Samza jobs represented by {@link StreamApplication}.
+   * Returns {@link ApplicationStatus#Running} if any of the jobs are running.
```

Comment:
Discussed offline. We can add it to ApplicationStatus when we have multi-stage jobs.